FifthLesson/VectorProgram/VectorProgram/Model/Vector.cs

```diff
@@ -1,115 +1,137 @@
+    public class Vector : IEquatable<Vector>
     {
-        public double CoordinateFirst { get; private set; }
-        public double CoordinateSecond { get; private set; }
-        public double CoordinateThird { get; private set; }
-
+        #region Properties
+        public double FirstCoordinate { get; private set; }
+        public double SecondCoordinate { get; private set; }
+        public double ThirdCoordinate { get; private set; }
         public double Length
         {
             get
             {
-                return Math.Round(Math.Sqrt(CoordinateFirst * CoordinateFirst + CoordinateSecond * CoordinateSecond + CoordinateThird * CoordinateThird), 3);
+                return Math.Round(Math.Sqrt(Math.Pow(FirstCoordinate, 2) + Math.Pow(SecondCoordinate, 2) + Math.Pow(ThirdCoordinate, 2)), 3);
             }
         }
+        #endregion
 
-        public Vector()
+        #region Construcrots
+        public Vector(double coordinateFirst, double coordinateSecond, double coordinateThird)
         {
-            CoordinateFirst = 0;
-            CoordinateSecond = 0;
-            CoordinateThird = 0;
+            FirstCoordinate = coordinateFirst;
+            SecondCoordinate = coordinateSecond;
+            ThirdCoordinate = coordinateThird;
         }
 
-        public Vector(double coordinateFirst, double coordinateSecond, double coordinateThird) : this()
-        {
-            CoordinateFirst = coordinateFirst;
-            CoordinateSecond = coordinateSecond;
-            CoordinateThird = coordinateThird;
-        }
+        public Vector() : this(0, 0, 0) { }
 
-        public Vector(Vector vector) : this()
-        {
-            CoordinateFirst = vector.CoordinateFirst;
-            CoordinateSecond = vector.CoordinateSecond;
-            CoordinateThird = vector.CoordinateThird;
-        }
+        public Vector(Vector vector) : this(vector.FirstCoordinate, vector.SecondCoordinate, vector.ThirdCoordinate) { }
+        #endregion
 
+        #region override ToString() GetHashCode() Equals() 
         public override string ToString()
         {
-            return $"({CoordinateFirst}, {CoordinateSecond}, {CoordinateThird})";
+            return $"({FirstCoordinate}, {SecondCoordinate}, {ThirdCoordinate})";
         }
 
-        public static Vector operator +(Vector vector1, Vector vector2)
+        public override int GetHashCode() => GetHashCodeHelper(
+               new int[]
+               {
+                   FirstCoordinate.GetHashCode(),
+                   SecondCoordinate.GetHashCode(),
+                   ThirdCoordinate.GetHashCode()
+               }
+            );
+
+        public override bool Equals(object obj) => this.Equals((Vector)obj);
+        #endregion
+
+        #region operators + - * == !=
+        public static Vector operator +(Vector first, Vector second)
         {
-            var result = new Vector(vector1);
+            var result = new Vector(first);
```

Comment:
It would be better to use object initializer instead:
_new Vector {
         FirstCoordinate = ...,
         ...
}_